src/model.rs

```diff
@@ -811,7 +811,7 @@ pub mod conv {
 
         fn from_str(tc: &str) -> Result<Tc> {
             use toolchain;
-            let _ = toolchain::parse_toolchain(tc)?;
+            let _ = tc.parse::<toolchain::Toolchain>()?;
             Ok(Tc(tc.to_string()))
```

Comment:
This is existing code, and save a bit of work, but I think it would be clearer if the code didn't throw away the toolchain, but rather used it to generate the string being returned.